spyder_kernels/customize/spydercustomize.py

```diff
@@ -783,13 +793,25 @@ def _get_globals():
     return ipython_shell.user_ns
 
 
+def _comms_enabled():
+    return get_ipython().kernel.frontend_comm.is_open()
+
+
+def _frontend_request():
+    return get_ipython().kernel.frontend_comm.remote_call(blocking=True)
+
+
 def runfile(filename, args=None, wdir=None, namespace=None, post_mortem=False):
     """
     Run filename
     args: command line arguments (string)
     wdir: working directory
     post_mortem: boolean, whether to enter post-mortem mode on error
     """
+    if _comms_enabled():
```

Comment:
This test is made for backward compatibility reasons. It can be removed if it is assumed that the comm must be open. Alternatively, The check could always be made for blocking calls, and non-blocking calls could be warnings.